common/djangoapps/student/models.py

```diff
@@ -3086,6 +3091,67 @@ def should_celebrate_first_section(enrollment):
             return False
 
 
+class UserCelebration(TimeStampedModel):
+    """
+    Keeps track of how we've celebrated a user's progress.
+
+    An example of a celebration is a dialog that pops up after you post for the first time in the
+    discussion forums.
+
+    In general, these celebrations should be course agnostic and could show up at any point for
+    a learner. Some will be one-off celebrations (like first discussion post) while others could
+    be repeated for a user (like streaks).
+    For celebrations that can be repeated with each enrollment, see CourseEnrollmentCelebration
+
+    .. no_pii:
+    """
+    user = models.OneToOneField(User, models.CASCADE, related_name='celebrations')
+    # Use a NullBooleanField in case another celebration creates the row.
```

Comment:
This line was confusing to me. I know what you mean, but at first, I was thinking "a different row creates this row"

Reply:
Do you think it is even necessary? Like do I need to explain why I'm using a NullBooleanField or should it just be a thing people figure out/know?

Reply:
Eh, I like some sort of comment here I guess. Maybe like "Null means we don't know yet, and have to ask the forums service"?